autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
I tried modifying things to set the signs in another buffer, but those changes were too deep and I felt beyond the scope of this PR. I am happy to only update the signs in the current buffer, I just don't want signs from another buffer showing up in my current one.

loclist, as I said, is already able to handle multiple files just fine.